cirq-core/cirq/value/classical_data.py

```diff
@@ -0,0 +1,278 @@
+            qubits: The qubits that were measured.
+
+        Raises:
+            ValueError: If the measurement shape does not match the qubits
+                measured or if the measurement key was already used.
+        """
+        if len(measurement) != len(qubits):
+            raise ValueError(f'{len(measurement)} measurements but {len(qubits)} qubits.')
+        if key in self._measurements:
+            raise ValueError(f"Measurement already logged to key {key}")
+        self._measurement_types[key] = MeasurementType.MEASUREMENT
+        self._measurements[key] = tuple(measurement)
+        self._measured_qubits[key] = tuple(qubits)
+
+    def record_channel_measurement(self, key: 'cirq.MeasurementKey', measurement: int):
+        """Records a channel measurement.
+
+        Args:
+            key: The measurement key to hold the measurement.
+            measurement: The measurement result.
+
+        Raises:
+            ValueError: If the measurement key was already used.
+        """
+        if key in self._measurement_types:
+            raise ValueError(f"Measurement already logged to key {key}")
+        self._measurement_types[key] = MeasurementType.CHANNEL
+        self._channel_measurements[key] = measurement
+
+    def get_digits(self, key: 'cirq.MeasurementKey') -> Tuple[int, ...]:
+        """Gets the digits of the measurement.
+
+        Args:
+            key: The measurement key.
+
+        Raises:
+            KeyError: If the key has not been used.
+        """
+        return (
+            self._measurements[key]
+            if self._measurement_types[key] == MeasurementType.MEASUREMENT
+            else (self._channel_measurements[key],)
+        )
+
+    def get_int(self, key: 'cirq.MeasurementKey') -> int:
+        """Gets the integer corresponding to the measurement.
+
+        Args:
+            key: The measurement key.
+
+        Raises:
+            KeyError: If the key has not been used.
+        """
+        if key not in self._measurement_types:
+            raise KeyError(f'The measurement key {key} is not in {self._measurements}')
+        measurement_type = self._measurement_types[key]
+        if measurement_type == MeasurementType.CHANNEL:
+            return self._channel_measurements[key]
+        if key not in self._measured_qubits:
+            return digits.big_endian_bits_to_int(self._measurements[key])
+        return digits.big_endian_digits_to_int(
+            self._measurements[key], base=[q.dimension for q in self._measured_qubits[key]]
+        )
+
+    def copy(self):
+        """Creates a copy of the object."""
+        return ClassicalDataDictionaryStore(
+            _measurements=self._measurements.copy(),
+            _measured_qubits=self._measured_qubits.copy(),
+            _channel_measurements=self._channel_measurements.copy(),
+            _measurement_types=self._measurement_types.copy(),
+        )
+
+    def _json_dict_(self):
```

Comment:
For consistent serialization behavior, these lists should be sorted.

Reply:
This was intentional. For python 3.6 up, dictionaries preserve insertion order, so this should be consistent. And I think the order of insertion would be useful to preserve on serialization too, especially since we expose a `keys` method.